### ISSUE TITLE
Add .png/jpg alternative for .svg images

### ISSUE THREAD

**Issue by DeD1rk** [svthalia/concrexit#1701]
### Describe the change
Since #1385, slides can be SVG. I suggest providing an alternative, such as having the thumbnail generator make png or jpg versions. 

### Motivation
Flutter does not support SVG's well yet (at least not the entire specification, see the screenshot) so this change would be nice/required to add slides to the app.

<img width="514" alt="image" src="https://user-images.githubusercontent.com/41264528/119850713-148a7e00-bf0e-11eb-86d8-91b0b7cf0b38.png">

I can imagine that this is quite a strange change, that might not be worth it.

**Comment by SebastiaanSafeguard** [comment]
I knew adding svg support would not be a good idea 😆

**Comment by se-bastiaan** [comment]
I knew it was a bad idea to support svg images.

Did you use https://pub.dev/packages/flutter_svg?

**Comment by DeD1rk** [comment]
> Did you use https://pub.dev/packages/flutter_svg?

yes, by itself using that package is something I'm not quite fond of (large package for almost no functionality), but apparently even with that not all SVGs are supported, and I don't think its really feasible to set any limits to what types of SVGs are allowed.

**Comment by se-bastiaan** [comment]
I'd be more in favour of reverting #1385 instead of adding a fallback.

**Comment by DeD1rk** [comment]
I tried using a webview as @pingiun suggested. That does show the svg correctly, but makes it very hard toward with the images: getting it in the right size is tricky and animations such as placeholders are virtually impossible. 

Given the advantage on large screens I suppose the best solution (other than just not adding slides to the app) would be to require the user to add a jpg/png fallback when they upload a jpg. The implementation should then probably be that the image can only be jpg/png and a user can optionally add an svg as well, and the website would then just show svg if it exists and jpg/png otherwise.

**Comment by DeD1rk** [comment]
For now I will filter out SVG slides client-side to simply not display them in the app. SVG support should be refactored to an optional additional field sometime.

**Comment by ColonelPhantom** [comment]
I guess letting the thumbnailer or similar convert the SVG to PNG's/JPG's would be a feasible solution, that way a client can ask for either a PNG/JPG version or SVG version.

I wouldn't mind looking into this, although I will probably need a bit of guidance.
